### PR TITLE
solve exp and difference of logs  with irrational coefficients

### LINKED ISSUE #15949: exp and difference of logs doesn't simplify with irrational coefficients
These expressions auto-simplify
```julia
In [36]: x, y = symbols('x y')                                                                                                    

In [37]: exp(log(x))                                                                                                              
Out[37]: x

In [38]: exp(log(x)-log(y))                                                                                                       
Out[38]: 
x
─
y

In [39]: exp(2*(log(x)-log(y)))                                                                                                   
Out[39]: 
 2
x 
──
 2
y 

In [41]: exp(sqrt(2)*log(x))                                                                                                      
Out[41]: 
 √2
x  
```
However this one does not simplify:
```julia
In [42]: exp(sqrt(2)*(log(x)-log(y)))                                                                                             
Out[42]: 
 √2⋅(log(x) - log(y))
ℯ                    

In [43]: simplify(_)                                                                                                              
Out[43]: 
 √2⋅(log(x) - log(y))
ℯ                    
```

I would expect one of these results:
```julia
In [44]: (x/y)**sqrt(2)                                                                                                           
Out[44]: 
   √2
⎛x⎞  
⎜─⎟  
⎝y⎠  

In [45]: x**sqrt(2)/y**sqrt(2)                                                                                                    
Out[45]: 
 √2  -√2
x  ⋅y   
```

### PR DESCRIPTION
**Pull request by RituRajSingh878**
#### References to other Issues or PRs

Fixes #15949

#### Brief description of what is fixed or changed
```
In [1]: x,y=symbols('x y',positive=True)                                                                                                                                                                     

In [2]: exp(sqrt(2)*(log(x)-log(y)))                                                                                                                                                                         
Out[2]: 
 √2⋅(log(x) - log(y))
ℯ    
```
As expression is not combining by itself.
So I use logcombine to combine when x,y is positive .
#### Release Notes

- functions
  - use logcombine to combine "logs  with irrational coefficients"

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
It's not clear to me what this change does. The example you've shown looks like the current behaviour.

Can you clearly show the difference between what happens on current master and what happens with this patch?

Also I'm sure you know by now that you should provide a test with something like this (note that the test helps to communicate what the patch does as well).

**Comment by RituRajSingh878** [comment]
In sympy master -
```
In [1]: x,y=symbols('x y',positive=True)                                                                                                                                                                     

In [2]: exp(sqrt(2)*(log(x)-log(y)))                                                                                                                                                                         
Out[2]: 
 √2⋅(log(x) - log(y))
ℯ    

```
Now it is working like-
```
In [1]: x,y=symbols('x y',positive=True)                                                                                                                                                                     

In [2]: exp(sqrt(2)*(log(x)-log(y)))                                                                                                                                                                         
Out[2]: 
   √2
⎛x⎞  
⎜─⎟  
⎝y⎠  
```

File: sympy/functions/elementary/exponential.py [hunk 1]
```diff
@@ -277,9 +278,9 @@ def eval(cls, arg):
 
             coeffs, log_term = [coeff], None
             for term in Mul.make_args(terms):
-                if isinstance(term, log):
```
Comment (RituRajSingh878) [inline sympy/functions/elementary/exponential.py]: @oscarbenjamin  you can look here-
In this line in sympy master it is getting (log(x)-log(y))(for which `isinstance(term, log)` is false) type of expression so I use logcombine  to check if there are any possibilities to make this type  ` log(exp)` .

File: sympy/functions/elementary/exponential.py [hunk 2]
```diff
@@ -277,9 +278,9 @@ def eval(cls, arg):
...
+                if isinstance(logcombine(term), log):
                     if log_term is None:
-                        log_term = term.args[0]
+                        log_term = logcombine(term).args[0]
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/exponential.py]: You are calling `logcombine` twice which is inefficient.

Comment (RituRajSingh878) [reply sympy/functions/elementary/exponential.py]: okk i got you.

**Comment by oscarbenjamin** [comment]
Should this not have had a test?

**Pull request merged**